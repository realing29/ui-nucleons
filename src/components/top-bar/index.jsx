import React from 'react';
import classnames from 'classnames/bind';
import PropTypes from 'prop-types';

import classes from './top-bar.scss';

const cx = classnames.bind(classes);

/**
 * Компонент шапки модальных окон/экранов.
 * @param {Object} props Свойства.
 * @param {'s'|'m'|'l'} [props.size='m'] Размер.
 * @param {string} props.title Заголовок.
 * @param {string} [props.subtitle] Подзаголовок.
 * @param {*} [props.startIcon] Начальная иконка.
 * @param {*} [props.endIcon] Конечная иконка.
 * @param {*} [props.startSecondaryIcon] Дополнительная начальная иконка (ближе к центру).
 * @param {*} [props.endSecondaryIcon] Дополнительная конечная иконка (ближе к центру).
 * @return {ReactElement} Компонент шапки модальных окон/экранов.
 */
const TopBar = ({
  size = 'm',
  title,
  subtitle,
  startIcon,
  endIcon,
  startSecondaryIcon,
  endSecondaryIcon,
}) => {
  const stub = startIcon || endIcon ? iconStub : null;
  const secondaryStub = startSecondaryIcon || endSecondaryIcon ? iconStub : null;

  return (
    <div className={cx('modal-header', `size-${size}`)}>
      {startIcon ? <IconButton children={startIcon} /> : stub}
      {startSecondaryIcon ? <IconButton children={startSecondaryIcon} /> : secondaryStub}

      <div className={cx('main-section')}>
        <div className={cx('title', 'ellipsis')}>{title}</div>
        {Boolean(subtitle) && (
          <div className={cx('subtitle', 'ellipsis')}>
            {subtitle}
          </div>
        )}
      </div>

      {endSecondaryIcon ? <IconButton children={endSecondaryIcon} /> : secondaryStub}
      {endIcon ? <IconButton children={endIcon} /> : stub}
    </div>
  );
};

TopBar.propTypes = {
  /**
   * Размер.
   */
  size: PropTypes.oneOf(['s', 'm', 'l']),

  /**
   * Заголовок.
   */
  title: PropTypes.string,

  /**
   * Подзаголовок.
   */
  subtitle: PropTypes.string,

  /**
   * Начальная иконка.
   */
  startIcon: PropTypes.any,

  /**
   * Конечная иконка.
   */
  endIcon: PropTypes.any,

  /**
   * Дополнительная начальная иконка (ближе к центру).
   */
  startSecondaryIcon: PropTypes.any,

  /**
   * Дополнительная конечная иконка (ближе к центру).
   */
  endSecondaryIcon: PropTypes.any,
};

/**
 * Компонент кнопки-иконки.
 * @param {Object} props Свойства. Поддерживаются свойства button.
 * @param {Object} props.children Содержимое.
 * @return {ReactElement} Компонент кнопки-иконки.
 */
export const IconButton = ({ children, ...props }) => (
  <button {...props} className={cx('icon-button')}>
    {children}
  </button>
);

IconButton.propTypes = {
  /**
   * Содержимое.
   */
  children: PropTypes.any,
};

const iconStub = (
  <div className={cx('icon-button')} />
);

export default TopBar;
