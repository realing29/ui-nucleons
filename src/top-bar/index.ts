export type { TopBarProps, TopBarSize } from './types';
export type { navigationButtons } from './utils';
export { TopBar, TOP_BAR_HEIGHT } from './top-bar';
