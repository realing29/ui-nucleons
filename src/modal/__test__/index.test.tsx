import React from 'react';
import { Modal } from '..';
import { mount, ReactWrapper } from 'enzyme';
import ArrowLeftSVG from '@dev-dep/ui-quarks/icons/24x24/Stroked/arrow-left';
import { enableBodyScroll, disableBodyScroll } from 'body-scroll-lock';
import { act } from 'react-dom/test-utils';
import { TopBar } from '../../top-bar';

jest.mock('body-scroll-lock', () => {
  const original = jest.requireActual('body-scroll-lock');

  return {
    ...original,
    __esModule: true,
    enableBodyScroll: jest.fn(original.enableBodyScroll),
    disableBodyScroll: jest.fn(original.disableBodyScroll),
  };
});

describe('<Modal />', () => {
  const findOverlay = (w: ReactWrapper) => w.find('[data-testid="modal:overlay"]');

  it('renders correct without close button and customClasses', () => {
    const wrapper = mount(<Modal>Test modal content</Modal>);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onClose properly', () => {
    const spy = jest.fn();

    const wrapper = mount(
      <Modal onClose={spy}>
        <Modal.Header title='Test title' onClose={spy} />
        <Modal.Body>
          Test modal content
        </Modal.Body>
      </Modal>
    );

    act(() => {
      findOverlay(wrapper).simulate('mousedown', { button: 0, target: 'B' });
      findOverlay(wrapper).simulate('mouseup', { button: 0, target: 'A', currentTarget: 'B' });
    });
    wrapper.update();
    expect(spy).toHaveBeenCalledTimes(0);

    act(() => {
      findOverlay(wrapper).simulate('mousedown', { button: 1, target: 'B' });
      findOverlay(wrapper).simulate('mouseup', { button: 1, target: 'A', currentTarget: 'B' });
    });
    wrapper.update();
    expect(spy).toHaveBeenCalledTimes(0);

    act(() => {
      findOverlay(wrapper).simulate('mousedown', { button: 0, target: 'A' });
      findOverlay(wrapper).simulate('mouseup', { button: 0, target: 'A', currentTarget: 'A' });
    });
    wrapper.update();
    expect(spy).toHaveBeenCalledTimes(0);

    act(() => {
      findOverlay(wrapper).simulate('mousedown', { button: 0, target: 'A' });
      findOverlay(wrapper).simulate('mouseup', { button: 0, target: 'A', currentTarget: 'B' });
    });
    wrapper.update();
    expect(spy).toHaveBeenCalledTimes(0);

    act(() => {
      const overlayEl = findOverlay(wrapper).getDOMNode();

      // с mount нельзя подменить currentTarget: https://github.com/enzymejs/enzyme/issues/1943
      findOverlay(wrapper).simulate('mousedown', { button: 0, target: overlayEl });
      findOverlay(wrapper).simulate('mouseup', { button: 0, target: overlayEl, currentTarget: overlayEl });
    });
    wrapper.update();
    expect(spy).toHaveBeenCalledTimes(1);

    act(() => {
      wrapper.find('button[data-testid="modal:close"]').simulate('click');
    });
    wrapper.update();
    expect(spy).toHaveBeenCalledTimes(2);
  });

  it('should handle extended props', () => {
    const wrapper = mount(
      <Modal>
        <Modal.Header title='Title' subtitle='Subtitle' />
        <Modal.Body>Main Content</Modal.Body>
        <Modal.Footer>
          <button>Hello</button>
        </Modal.Footer>
      </Modal>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should use disable/enable body scrolling', () => {
    const wrapper = mount(
      <Modal />
    );

    expect(disableBodyScroll).toHaveBeenCalledTimes(1);
    expect(enableBodyScroll).toHaveBeenCalledTimes(0);

    wrapper.unmount();

    expect(disableBodyScroll).toHaveBeenCalledTimes(1);
    expect(enableBodyScroll).toHaveBeenCalledTimes(1);
  });

  it('should handle "withScrollDisable" and "scrollDisableOptions" options', () => {
    const wrapper = mount(
      <Modal
        withScrollDisable
        scrollDisableOptions={{}}
      />
    );

    expect(disableBodyScroll).toHaveBeenCalledTimes(1);
    expect(enableBodyScroll).toHaveBeenCalledTimes(0);

    wrapper.unmount();

    expect(disableBodyScroll).toHaveBeenCalledTimes(1);
    expect(enableBodyScroll).toHaveBeenCalledTimes(1);
  });

  it('should render different sizes properly', () => {
    (['s', 'm', 'l', 'xl', 'fullscreen'] as const).forEach(size => {
      const wrapper = mount(
        <Modal size={size} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should renders in layer', () => {
    const wrapper = mount(
      <Modal withLayer />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render TopBar with back button', () => {
    const spy = jest.fn();

    const wrapper = mount(
      <Modal>
        <Modal.Header onBack={spy} />
      </Modal>
    );

    expect(wrapper).toMatchSnapshot();

    expect(spy).toBeCalledTimes(0);

    act(() => {
      wrapper.find('button[data-testid="modal:back"]').simulate('click');
    });
    wrapper.update();

    expect(spy).toBeCalledTimes(1);
  });

  it('should render ith close and custom start button in top bar', () => {
    const wrapper = mount(
      <Modal>
        <Modal.Header
          onClose={jest.fn()}
          buttonsProps={{
            start: {
              icon: <ArrowLeftSVG data-testid='custom-top-bar-btn' />,
            },
          }}
        />
      </Modal>
    );

    expect(wrapper.find(TopBar).find('button[data-testid="modal:close"]')).toHaveLength(1);
    expect(wrapper.find(TopBar).find('svg[data-testid="custom-top-bar-btn"]')).toHaveLength(1);
  });

  it('should handle "height" prop', () => {
    const wrapper = mount(
      <Modal height={480} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should renders with fullscreen size and footer', () => {
    const wrapper = mount(
      <Modal size='fullscreen'>
        <Modal.Footer>Test footer</Modal.Footer>
      </Modal>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
