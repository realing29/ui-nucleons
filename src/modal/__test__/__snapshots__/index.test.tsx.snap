// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correct without close button and customClasses 1`] = `
<Modal>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-m shadow-z4"
      data-testid="modal"
    >
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                >
                  Test modal content
                </div>
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should handle extended props 1`] = `
<Modal
  footer={
    <button>
      Hello
    </button>
  }
  onClose={[MockFunction]}
  subtitle="Subtitle"
  title="Title"
  topBarProps={
    Object {
      "size": "m",
    }
  }
  withCloseButton={true}
  withDivideFooter={true}
  withDivideTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-m shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={
          Object {
            "end": Object {
              "data-testid": "modal:close",
              "icon": <CrossSVG />,
              "onClick": [MockFunction],
            },
          }
        }
        className="header inner-border-bottom"
        size="s"
        subtitle="Subtitle"
        title="Title"
      >
        <div
          className="root size-s header inner-border-bottom"
        >
          <div
            className="icon-button stub"
          />
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            >
              Title
            </div>
            <div
              className="subtitle ellipsis"
            >
              Subtitle
            </div>
          </div>
          <IconButton
            data-testid="modal:close"
            icon={<CrossSVG />}
            onClick={[MockFunction]}
          >
            <button
              className="icon-button"
              data-testid="modal:close"
              onClick={[MockFunction]}
              type="button"
            >
              <CrossSVG>
                <svg
                  height={24}
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <path
                    clipRule="evenodd"
                    d="M4.293 19.707a1 1 0 001.414 0L12 13.414l6.293 6.293a1 1 0 001.414-1.414L13.414 12l6.293-6.293a1 1 0 00-1.414-1.414L12 10.586 5.707 4.293a1 1 0 00-1.414 1.414L10.586 12l-6.293 6.293a1 1 0 000 1.414z"
                    fillRule="evenodd"
                  />
                </svg>
              </CrossSVG>
            </button>
          </IconButton>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer inner-border-top"
      >
        <button>
          Hello
        </button>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should handle extended props 2`] = `
<Modal
  footer={
    <button>
      Hello
    </button>
  }
  onClose={[MockFunction]}
  subtitle="Subtitle"
  title="Title"
  topBarProps={
    Object {
      "size": "m",
    }
  }
  withCloseButton={false}
  withDivideFooter={true}
  withDivideTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-m shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={Object {}}
        className="header inner-border-bottom"
        size="s"
        subtitle="Subtitle"
        title="Title"
      >
        <div
          className="root size-s header inner-border-bottom"
        >
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            >
              Title
            </div>
            <div
              className="subtitle ellipsis"
            >
              Subtitle
            </div>
          </div>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer inner-border-top"
      >
        <button>
          Hello
        </button>
      </div>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should render TopBar with back button 1`] = `
<Modal
  onBack={[MockFunction]}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-m shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={
          Object {
            "start": Object {
              "data-testid": "modal:back",
              "icon": <ArrowLeftSVG />,
              "onClick": [MockFunction],
            },
          }
        }
        className="header"
        size="s"
      >
        <div
          className="root size-s header"
        >
          <IconButton
            data-testid="modal:back"
            icon={<ArrowLeftSVG />}
            onClick={[MockFunction]}
          >
            <button
              className="icon-button"
              data-testid="modal:back"
              onClick={[MockFunction]}
              type="button"
            >
              <ArrowLeftSVG>
                <svg
                  height={24}
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <path
                    d="M5.414 13l5.293 5.293a1 1 0 01-1.414 1.414l-7-7a1 1 0 010-1.414l7-7a1 1 0 011.414 1.414L5.414 11H21a1 1 0 110 2H5.414z"
                  />
                </svg>
              </ArrowLeftSVG>
            </button>
          </IconButton>
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            />
          </div>
          <div
            className="icon-button stub"
          />
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should render different sizes properly 1`] = `
<Modal
  size="s"
  withTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-s shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={Object {}}
        className="header"
        size="s"
      >
        <div
          className="root size-s header"
        >
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            />
          </div>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should render different sizes properly 2`] = `
<Modal
  size="m"
  withTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-m shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={Object {}}
        className="header"
        size="s"
      >
        <div
          className="root size-s header"
        >
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            />
          </div>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should render different sizes properly 3`] = `
<Modal
  size="l"
  withTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-l shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={Object {}}
        className="header"
        size="s"
      >
        <div
          className="root size-s header"
        >
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            />
          </div>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should render different sizes properly 4`] = `
<Modal
  size="xl"
  withTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-xl shadow-z4"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={Object {}}
        className="header"
        size="s"
      >
        <div
          className="root size-s header"
        >
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            />
          </div>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should render different sizes properly 5`] = `
<Modal
  size="fullscreen"
  withTopBar={true}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-fullscreen"
      data-testid="modal"
    >
      <TopBar
        buttonsProps={Object {}}
        className="header"
        size="m"
      >
        <div
          className="root size-m header"
        >
          <div
            className="main-section"
          >
            <div
              className="title ellipsis"
            />
          </div>
        </div>
      </TopBar>
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
    </div>
  </div>
</Modal>
`;

exports[`<Modal /> should renders in layer 1`] = `
<Modal
  withLayer={true}
>
  <Layer>
    <Portal
      containerInfo={
        <div>
          <div
            class="overlay"
            data-testid="modal:overlay"
          >
            <div
              class="modal size-m shadow-z4"
              data-testid="modal"
            >
              <div
                class="os-host os-host-foreign os-theme-dark os-host-resize-disabled os-host-scrollbar-horizontal-hidden os-host-scrollbar-vertical-hidden custom-scrollbar main"
              >
                <div
                  class="os-resize-observer-host"
                >
                  <div
                    class="os-resize-observer"
                    style="right: 0px;"
                  >
                    <div
                      class="os-resize-observer-item"
                      dir="ltr"
                      style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
                    >
                      <div
                        class="os-resize-observer-item"
                      >
                        <div
                          class="os-resize-observer-item-final"
                          style="width: 3333333px; height: 3333333px;"
                        />
                      </div>
                      <div
                        class="os-resize-observer-item"
                      >
                        <div
                          class="os-resize-observer-item-final"
                          style="width: 200%; height: 200%"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="os-size-auto-observer"
                  style="height: calc(100% + 1px); float: left;"
                >
                  <div
                    class="os-resize-observer"
                  >
                    <div
                      class="os-resize-observer-item"
                      dir="ltr"
                      style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
                    >
                      <div
                        class="os-resize-observer-item"
                      >
                        <div
                          class="os-resize-observer-item-final"
                          style="width: 3333333px; height: 3333333px;"
                        />
                      </div>
                      <div
                        class="os-resize-observer-item"
                      >
                        <div
                          class="os-resize-observer-item-final"
                          style="width: 200%; height: 200%"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="os-content-glue"
                  style="margin: 0px 0px 0px 0px; width: -1px; height: -1px;"
                />
                <div
                  class="os-padding"
                >
                  <div
                    class="os-viewport os-viewport-native-scrollbars-overlaid"
                    style=""
                  >
                    <div
                      class="os-content-arrange"
                    />
                    <div
                      class="os-content"
                      style="padding: 0px 0px 0px 0px; height: 100%; width: 100%;"
                    />
                  </div>
                </div>
                <div
                  class="os-scrollbar os-scrollbar-horizontal os-scrollbar-unusable"
                >
                  <div
                    class="os-scrollbar-track os-scrollbar-track-off"
                  >
                    <div
                      class="os-scrollbar-handle"
                      style="transform: translate(0px, 0);"
                    />
                  </div>
                </div>
                <div
                  class="os-scrollbar os-scrollbar-vertical os-scrollbar-unusable"
                >
                  <div
                    class="os-scrollbar-track os-scrollbar-track-off"
                  >
                    <div
                      class="os-scrollbar-handle"
                      style="transform: translate(0, 0px);"
                    />
                  </div>
                </div>
                <div
                  class="os-scrollbar-corner"
                />
              </div>
              <div
                class="footer-stub"
              />
            </div>
          </div>
        </div>
      }
    >
      <div
        className="overlay"
        data-testid="modal:overlay"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <div
          className="modal size-m shadow-z4"
          data-testid="modal"
        >
          <CustomScrollbar
            className="main"
            overflow={
              Object {
                "x": "h",
                "y": "s",
              }
            }
          >
            <OverlayScrollbarsComponent
              className="custom-scrollbar main"
              options={
                Object {
                  "overflowBehavior": Object {
                    "x": "h",
                    "y": "s",
                  },
                }
              }
            >
              <div
                className="os-host"
              >
                <div
                  className="os-resize-observer-host"
                />
                <div
                  className="os-padding"
                >
                  <div
                    className="os-viewport"
                  >
                    <div
                      className="os-content"
                    />
                  </div>
                </div>
                <div
                  className="os-scrollbar os-scrollbar-horizontal "
                >
                  <div
                    className="os-scrollbar-track"
                  >
                    <div
                      className="os-scrollbar-handle"
                    />
                  </div>
                </div>
                <div
                  className="os-scrollbar os-scrollbar-vertical"
                >
                  <div
                    className="os-scrollbar-track"
                  >
                    <div
                      className="os-scrollbar-handle"
                    />
                  </div>
                </div>
                <div
                  className="os-scrollbar-corner"
                />
              </div>
            </OverlayScrollbarsComponent>
          </CustomScrollbar>
          <div
            className="footer-stub"
          />
        </div>
      </div>
    </Portal>
  </Layer>
</Modal>
`;

exports[`<Modal /> should renders without TopBar 1`] = `
<Modal
  withTopBar={false}
>
  <div
    className="overlay"
    data-testid="modal:overlay"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="modal size-m shadow-z4"
      data-testid="modal"
    >
      <CustomScrollbar
        className="main"
        overflow={
          Object {
            "x": "h",
            "y": "s",
          }
        }
      >
        <OverlayScrollbarsComponent
          className="custom-scrollbar main"
          options={
            Object {
              "overflowBehavior": Object {
                "x": "h",
                "y": "s",
              },
            }
          }
        >
          <div
            className="os-host"
          >
            <div
              className="os-resize-observer-host"
            />
            <div
              className="os-padding"
            >
              <div
                className="os-viewport"
              >
                <div
                  className="os-content"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-horizontal "
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar os-scrollbar-vertical"
            >
              <div
                className="os-scrollbar-track"
              >
                <div
                  className="os-scrollbar-handle"
                />
              </div>
            </div>
            <div
              className="os-scrollbar-corner"
            />
          </div>
        </OverlayScrollbarsComponent>
      </CustomScrollbar>
      <div
        className="footer-stub"
      />
    </div>
  </div>
</Modal>
`;
